src/python/gudhi/random_point_generators.cc

```diff
@@ -0,0 +1,63 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hind Montassif
+ *
+ *    Copyright (C) 2021 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+
+#include <gudhi/random_point_generators.h>
+
+#include <CGAL/Epick_d.h>
+
+namespace py = pybind11;
+
+
+typedef CGAL::Epick_d< CGAL::Dynamic_dimension_tag > Kern;
+
+template <typename Kernel>
+py::array_t<double> generate_points_on_sphere(size_t num_points, int dim, double radius) {
```

Comment:
If I am using a CGAL::Epeck_d as Kernel, this template function is not compiling.
I think we could make this function non-template and use the kernel type definition inside it.
Or maybe the idea was to define it externally to be re-used for the other functions that will come thereafter ?

Reply:
Apart from Delaunay_complex_unit_test, the generate_points_on_sphere_d function doesn't seem to be used other than with CGAL::Epick_d kernel. Do we want to keep it that way ? In this case, no need for a template function indeed.
@mglisse ?

Reply:
generate_points_on_sphere_d could be used with different kernels (in particular Epick_d with different dimension parameters), so a template makes sense. generate_points_on_sphere_d is private in a .cc file and only instantiated once, so indeed it does not need to be a template.